### PR TITLE
Dispose of detectors correctly

### PR DESCRIPTION
**Pull request by melutzky**

### CODE REVIEW

File: test/analysis/SmartDetectorsAnalysisTests/SmartDetectorRunnerTests.cs [hunk 1]
```diff
@@ -320,5 +338,15 @@ public TestAlert(ResourceIdentifier resourceIdentifier)
             [AlertPresentationProperty(AlertPresentationSection.Property, "Summary title", InfoBalloon = "Summary info")]
             public string Summary { get; } = "Summary value";
         }
+
+        public sealed class TestSmartDetectorDisposable : TestSmartDetector, IDisposable
```
Comment (EranGon) [inline test/analysis/SmartDetectorsAnalysisTests/SmartDetectorRunnerTests.cs]: >TestSmartDetectorDisposable [](start = 28, length = 27)

DisposableTestSmartDetector

File: test/analysis/SmartDetectorsAnalysisTests/SmartDetectorRunnerTests.cs [hunk 2]
```diff
@@ -320,5 +338,15 @@ public TestAlert(ResourceIdentifier resourceIdentifier)
...
+        {
+            public bool Disposed { get; private set; }
```
Comment (EranGon) [inline test/analysis/SmartDetectorsAnalysisTests/SmartDetectorRunnerTests.cs]: >Disposed [](start = 24, length = 8)

WasDisposed

Comment (EranGon) [reply test/analysis/SmartDetectorsAnalysisTests/SmartDetectorRunnerTests.cs]: and initialize with true in the ctor

---
In reply to: [232665250](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/21#discussion_r232665250) [](ancestors = 232665250)

Comment (melutzky) [reply test/analysis/SmartDetectorsAnalysisTests/SmartDetectorRunnerTests.cs]: ctor = constructor, right?

---
In reply to: [232665330](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/21#discussion_r232665330) [](ancestors = 232665330,232665250)

**Pull request merged**